libs/wingsdk/src/core/internal.ts

```diff
@@ -69,13 +69,38 @@ ${Object.entries(clients)
   return new Handler();
 }
 
-function serializeImmutableData(obj: any): Code {
-  // if this is a Set, we need to convert it to an array and reconstruct on the other side
-  if (typeof obj === "object" && (obj as any) instanceof Set) {
-    return NodeJsCode.fromInline(`new Set(${JSON.stringify(Array.from(obj))})`);
+function serializeImmutableData(obj: any): string {
+  switch (typeof obj) {
+    case "string":
+    case "boolean":
+    case "number":
+      return JSON.stringify(obj);
+
+    case "object":
+      if (Array.isArray(obj)) {
+        return JSON.stringify(obj);
```

Comment:
I think this should be a recursive call to `serializeImmutableData` (what if it's an Array of Maps etc.)

Reply:
🤦 of course!

But this naturally brings up the fact that separating `data` and `resources` is artificial and won't hold water because it should be absolutely possible to capture
an array of resources:

```js
let arr = [new cloud.Bucket(), new cloud.Bucket()];
let handler = inflight () => {
  for b in arr {
    b.put("file.txt", "hello");
  }
});
```

So we need a unified serialization model.

I'll open a bug and work on this in a subsequent commit.

Reply:
https://github.com/winglang/wing/issues/1156

Reply:
I see, you're totally right. Sounds like good merit for joining them 👍

Reply:
Updated the code to serialize data recursively